MailCore/Cache/Actions/ActionsManager.swift

```diff
@@ -0,0 +1,245 @@
+/*
+ Infomaniak Mail - iOS App
+ Copyright (C) 2022 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import InfomaniakCoreUI
+import InfomaniakDI
+import MailResources
+import SwiftUI
+
+extension [Message]: Identifiable {
+    public var id: Int {
+        // Calculate a unique identifier by XORing hash values of messages
+        return reduce(1) { $0.hashValue ^ $1.hashValue }
```

Comment:
I double checked, the order of XOR operations does not affect the final result when you're XORing multiple values together. So this works well actualy, as I think we care about having a stable ID for a `Set` (regardless of ordering) of messages which this does.

Reply:
It looks like we use it for identity of swiftUI views, therefore it might be a bug for `[MessageA, MessageB]` and `[MessageB, MessageA]` to have the same identity 🤔

Reply:
tl;dr you can resolve this @PhilippeWeidmann. We only need to keep in mind that if we used a `[Message]` directly in a swiftUI List it will probably not refresh on order change.